common/transport/amqp/test/_amqp_cbs_test.js

```diff
@@ -27,60 +27,66 @@ describe('ClaimsBasedSecurityAgent', function() {
-      fakeReceiver.detach = sinon.stub().resolves();
-      fakeReceiver.forceDetach = function () {};
-      sinon.spy(fakeReceiver, 'on');
-      sinon.spy(fakeSender, 'on');
-      fakeAmqpClient.createSender = sinon.stub().resolves(fakeSender);
-      fakeAmqpClient.createReceiver = sinon.stub().resolves(fakeReceiver);
-
-      var cbs = new CBS(fakeAmqpClient);
+      var fakeRheaSession = new EventEmitter();
+      var cbs = new CBS(fakeRheaSession);
+      var fakeRheaLink = new EventEmitter();
+      var fakeContext = {receiver: fakeRheaLink};
+      fakeRheaSession.open_receiver = () => {};
+      sinon.stub(fakeRheaSession, 'open_receiver').callsFake(() => {process.nextTick( () => {fakeRheaLink.emit('receiver_open', fakeContext)});return fakeRheaLink;});
+      cbs._senderLink.attach = sinon.stub().callsArg(0);
+      cbs._senderLink.detach = sinon.stub().callsArg(0);
+      cbs._receiverLink.on = sinon.spy();
+      cbs._receiverLink.attach = sinon.stub().callsArgWith(0);
+      cbs._receiverLink.detach = sinon.stub().callsArgWith(0);
+
       cbs.attach(function(err) {
         /*Tests_SRS_NODE_AMQP_CBS_16_003: [`attach` shall attach the sender link.]*/
-        assert(fakeAmqpClient.createSender.calledOnce);
+        assert(cbs._senderLink.attach.calledOnce);
         /*Tests_SRS_NODE_AMQP_CBS_16_004: [`attach` shall attach the receiver link.]*/
-        assert(fakeAmqpClient.createReceiver.calledOnce);
+        assert(cbs._receiverLink.attach.calledOnce);
         /*Tests_SRS_NODE_AMQP_CBS_16_005: [The `attach` method shall set up a listener for responses to put tokens on the `message` event of the receiver link.]*/
-        assert(fakeReceiver.on.calledWith('message'));
+        assert(cbs._receiverLink.on.calledWith('message'));
         assert.isUndefined(err);
         testCallback();
       });
     });
 
     /*Tests_SRS_NODE_AMQP_CBS_16_007: [If given as an argument, the `attach` method shall call `callback` with a `null` error object if successful.]*/
     it('calls the callback immediately if links are already attached', function(testCallback) {
-      var fakeAmqpClient = new EventEmitter();
-      fakeAmqpClient.createReceiver = sinon.stub().resolves(new EventEmitter()),
-      fakeAmqpClient.createSender = sinon.stub().resolves(new EventEmitter())
+      var fakeRheaSession = new EventEmitter();
+
+      var cbs = new CBS(fakeRheaSession);
+
+      var fakeRheaLink = new EventEmitter();
+      var fakeContext = {receiver: fakeRheaLink};
+      fakeRheaSession.open_receiver = () => {};
+      sinon.stub(fakeRheaSession, 'open_receiver').callsFake(() => {process.nextTick( () => {fakeRheaLink.emit('receiver_open', fakeContext)});return fakeRheaLink;});
+      cbs._senderLink.attach = sinon.stub().callsArg(0);
+      cbs._senderLink.detach = sinon.stub().callsArg(0);
+      cbs._receiverLink.attach = sinon.stub().callsArgWith(0);
+      cbs._receiverLink.detach = sinon.stub().callsArgWith(0);
```

Comment:
a lot of this seems to repeat. I would suggest creating a stubLinks(cbs) helper function and just call this everywhere. then override the default stubs only when necessary.